src/AggregateRoots/AggregateRoot.php

```diff
@@ -29,6 +30,35 @@ abstract class AggregateRoot
 
     protected static bool $allowConcurrency = false;
 
+    /** @var \Spatie\EventSourcing\AggregateRoots\AggregateEntity[] */
+    private array $entities = [];
+
+    public function __construct()
```

Comment:
Pretty cool that we can automatically new up these entities 👍

Reply:
Seemed like the right thing to do. Unfortunately it also means the entity properties should be protected. If you want to keep them private you should manually set them up, but I wouldn't advice to do that.